src/Microsoft.Tye.Hosting/Model/Service.cs

```diff
@@ -17,12 +17,15 @@ public Service(ServiceDescription description)
 
             Logs.Subscribe(entry =>
             {
-                if (CachedLogs.Count > 5000)
+                lock (CachedLogs)
                 {
-                    CachedLogs.Dequeue();
-                }
+                    if (CachedLogs.Count > 5000)
+                    {
+                        CachedLogs.Dequeue();
+                    }
 
-                CachedLogs.Enqueue(entry);
+                    CachedLogs.Enqueue(entry);
+                }
```

Comment:
This doesn't seem like it goes far enough. CachedLogs isn't thread-safe and there's nothing ensuring *readers* won't see torn state (it's public).

The locking aspect seems find, but it seems like this should ConcurrentQueue as well or something else thread-safe for a reader.

Reply:
Yep, np. This is all janky stuff for the dashboard...